src/modules/surjection/main_impl.h

```diff
@@ -322,6 +322,10 @@ int secp256k1_surjectionproof_generate(const secp256k1_context* ctx, secp256k1_s
     /* Compute public keys */
     n_total_pubkeys = secp256k1_surjectionproof_n_total_inputs(ctx, proof);
     n_used_pubkeys = secp256k1_surjectionproof_n_used_inputs(ctx, proof);
+
+    /* This must be true if the proof was created with surjectionproof_initialize */
+    ARG_CHECK(n_used_pubkeys > 0);
+
     if (n_used_pubkeys > n_total_pubkeys || n_total_pubkeys != n_ephemeral_input_tags) {
         return 0;
     }
```

Comment:
Is it inconsistent to return different errors here?

Should the ARG_CHECK be moved up to the other ARG_CHECKs?

Reply:
`ARG_CHECK` just seems like the right thing to do (since it's an API violation) and I didn't want to touch existing code (which doesn't have tests afaics).

It can't be moved up because we only set n_used_pubkeys in the line right above this new ARG_CHECK.

Reply:
> `ARG_CHECK` just seems like the right thing to do (since it's an API violation) and I didn't want to touch existing code (which doesn't have tests afaics).

Hm yeah, I assumed this may contradict some some unwritten rule that ARG_CHECKs checks things that should be enforceable by normal type system. But in this case, rust-secp256k1-zkp anyway always calls `initialize` and `create` in a row. (This makes me wonder why these are separate functions in C.)

Reply:
> It can't be moved up because we only set n_used_pubkeys in the line right above this new ARG_CHECK.

Ok, I meant it could be moved together together with the other code, because it looked wrong on first glance to have `ARG_CHECK`s, then `return 0` a few times , and then an `ARG_CHECK` again. 

But ok, it's not wrong. I still think it will be cleaner to have all the `ARG_CHECK` first.

Reply:
We should look at merging `initialize` and `create`. I think this was my doing and I don't remember what my motivation was.

Reply:
@real-or-random 

> Ok, I meant it could be moved together together with the other code,

Ah sorry, got it. Didn't realize I can just move the other code to the top. Updated the commit.